src/core/model_abstract.jl

```diff
@@ -172,11 +172,11 @@ Return a valid `id` for creating a new agent with it.
 nextid(model::ABM) = notimplemented(model)
 
 """
-    add_agent_to_model!(agent, model)
+    add_agent_to_container!(agent, model)
 Add the agent to the model's internal container, if the addition is valid
 given the agent's ID and those already in the model. Otherwise error.
 """
-add_agent_to_model!(agent, model) = notimplemented(model)
+add_agent_to_container!(agent, model) = notimplemented(model)
```

Comment:
I think this function should dispatch to `add_agent_to_container!(agent, agent_container(model))`. 

I am not convinced we need a `notimplemented` function here. The only way this function is called is anyways in the high level.